### PR TITLE
[ui] Repair scroll behavior on dynamic virtualized lists

### LINKED ISSUE #17892: Code locations page scrambled
### Dagster version

1.5.7

### What's the issue?

Updated from 1.4.16 to 1.5.7 and found that the Code locations page layout is a bit scrambled now.
The locations table height is very small hiding some bottom parts (showing a vertical scroll-bar where it is not needed) and the row-columns are not in line with the header-columns.
I could make an screenshot but don´t know how to post it here.

### What did you expect to happen?

Sound page table-layout.

### How to reproduce?

_No response_

### Deployment type

Other

### Deployment details

_No response_

### Additional information

_No response_

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

### PR DESCRIPTION
**Pull request by hellendag**
## Summary & Motivation

Resolves #17892.

Repair a bug in virtualized dynamic-height lists in which the subpixel row height is causing an extra scrollbar to appear, specifically in the code locations list. The row heights are rounded for the summed container height, which can lead to the container being just a bit shorter than the total rendered row height, leading to an unwanted scrollbar.

Worse, the scrollbar sometimes broke the grid layout, causing the header column display and row column displays to become misaligned.

In this case, most notably, I needed to remove the `overflow: hidden` wrapper div. This way, even though the container may have a height that is less than the actual rendered row height sum, no scrollbar appears.

While working on this, I upgraded to v3.0.1 of the tanstack virtualizer lib, which required an update to the API usage. Specifically, `measureElement` is now on the `virtualizer` object, and the individual rows need a `data-key` prop.

## How I Tested These Changes

View code locations list, with the `count` value restricted to a smaller number. Verify that the unwanted scrollbar does not appear, even though the container height is less than the dynamically rendered row height sum.

View the side nav in the global asset graph, verify that it still renders and behaves correctly.

### CODE REVIEW

**Comment by hellendag** [comment]
Current dependencies on/for this PR:
* master
  * **PR #18564** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18564?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/18564?utm_source=stack-comment).

File: js_modules/dagster-ui/packages/ui-core/src/workspace/VirtualizedCodeLocationRow.tsx [hunk 1]
```diff
@@ -30,54 +30,57 @@ const TEMPLATE_COLUMNS = '3fr 1fr 1fr 240px 160px';
...
-            <ReloadButton location={name} />
-            <CodeLocationMenu locationNode={locationNode} />
-          </JoinedButtons>
-        </RowCell>
-      </RowGrid>
-    </div>
-  );
-};
+export const VirtualizedCodeLocationErrorRow = React.forwardRef(
+  (props: ErrorRowProps, ref: any) => {
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/workspace/VirtualizedCodeLocationRow.tsx:37]: I should see if I can get a better `ref` type here.

Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/workspace/VirtualizedCodeLocationRow.tsx:37]: I think you want `React.ForwardedRef<HTMLDivElement>`?

File: js_modules/dagster-ui/packages/ui-core/src/workspace/RepositoryLocationsList.tsx [hunk 2]
```diff
@@ -70,35 +70,35 @@ export const RepositoryLocationsList = ({loading, codeLocations, searchValue}: P
   return (
     <>
       <VirtualizedCodeLocationHeader />
-      <div style={{overflow: 'hidden'}}>
```
Comment (bengotow) [inline js_modules/dagster-ui/packages/ui-core/src/workspace/RepositoryLocationsList.tsx:73]: It's sorta interesting that this fixes the problem, I sort of think of `overflow:hidden` == no scroll bars. Maybe removing the container div entirely had some other impact too? (Either way, I'm glad you got to the bottom of this, what a whacky bug)

**Review by bengotow — APPROVED** [review]

**Comment by hellendag** [comment]
Jest is being annoying about the virtual list. Trying to fix.

**Pull request merged**